File: smart-english-backend/src/main/java/com/smartenglishbackend/esentity/Article.java
package com.smartenglishbackend.esentity;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;
import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;

import java.time.LocalDate;
import java.util.List;

@Data
@Document(indexName="articles")
public class Article {
    @Id
    @Field(name="_id")
    private String id;
    private String title;
    private String cover;
    private String content;
    @Field(type = FieldType.Date)
    private LocalDate date;
    private List<String> tags;
}

File: Mobile/app/src/main/java/site/smartenglish/remote/data/Article.kt
package site.smartenglish.remote.data

typealias SearchArticleResponse = List<SearchArticleResponseElement?>?

data class GetArticleResponse(
    val content: String? = null,
    val cover: String? = null,
    val date: String? = null,
    val id: String? = null,
    val tags: List<String?>? = null,
    val title: String? = null
)

data class SearchArticleResponseElement(
    val cover: String? = null,
    val date: String? = null,
    val id: String? = null,
    val tags: List<String?>? = null,
    val title: String? = null
)